src/github.com/mozilla-services/pushgo/simplepush/worker.go

```diff
@@ -136,35 +140,47 @@ func harmlessConnectionError(err error) (harmless bool) {
 	return
 }
 
+func (self *WorkerWS) Deadline() (t time.Time) {
+	var d time.Duration
+	switch self.state {
+	case WorkerInactive:
+		d = self.helloTimeout
+	case WorkerActive:
+		d = self.pongInterval
+	}
+	if d > 0 {
+		t = timeNow().Add(d)
+	}
```

Comment:
Yup; in that case, no deadline will be set. Good candidate for a doc string. :grinning: